targets/psa_switch/psa_switch.cpp

```diff
@@ -396,6 +396,20 @@ PsaSwitch::egress_thread(size_t worker_id) {
     Pipeline *egress_mau = this->get_pipeline("egress");
     egress_mau->apply(packet.get());
     deparser->deparse(packet.get());
+    
+    if (port == PSA_PORT_RECIRCULATE) {
+      BMLOG_DEBUG_PKT(*packet, "Recirculating packet");
+      std::unique_ptr<Packet> packet_copy = packet->clone_no_phv_ptr();
+      PHV *phv_copy = packet_copy->get_phv();
+      phv_copy->reset_metadata();
+      phv_copy->get_field("psa_ingress_input_metadata.packet_path")
```

Comment:
As mentioned in a previous PR, we should be using the PACKET_PATH enum from JSON. Compiler support is currently pending approval on p4c PR.

Reply:
I think there are 2 different packet_path metadata fields that must be assigned here, the one you have already, plus the one in `psa_ingress_parser_input_metadata`

Reply:
The PSA spec also says that recirculated packets should have both of these fields assigned the value of PSA_RECIRCULATE_PORT for recirculated packets.  That should be done somewhere near this line of code, too:

+ psa_ingress_parser_input_metadata.ingress_port
+ psa_ingress_input_metadata..ingress_port

Reply:
Another field: psa_ingress_input_metadata.ingress_timestamp should have the current time stored into it when a packet is recirculated, according to the PSA spec.